airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/AbstractJdbcSource.java

```diff
@@ -410,6 +414,29 @@ public JdbcDatabase createDatabase(final JsonNode sourceConfig) throws SQLExcept
     quoteString = (quoteString == null ? database.getMetaData().getIdentifierQuoteString() : quoteString);
     database.setSourceConfig(sourceConfig);
     database.setDatabaseConfig(jdbcConfig);
+    LOGGER.info("Data source product recognized as {}:{}",
+        database.getMetaData().getDatabaseProductName(),
+        database.getMetaData().getDatabaseProductVersion());
+    final ResultSet tables = database.getMetaData().getTables(null,
+        null,
+        "%",
+        new String[] {JDBC_TABLE});
+    while (tables.next()) {
+      final String tableName = tables.getString(JDBC_COLUMN_TABLE_NAME);
+      final ResultSet indexInfo = database.getMetaData().getIndexInfo(null,
+          null,
+          tableName,
+          false,
+          false);
+      LOGGER.info("Discovering indexes for table \"{}\"", tableName);
+      while (indexInfo.next()) {
+        LOGGER.info("Index name: {}, Column: {}, Unique: {}",
+            indexInfo.getString(JDBC_INDEX_NAME),
```

Comment:
Unfortunately, since JDBC drivers differ quite a bit in how they implement methods, you might have do null checks every time you're checking a field. see : `AbstractJdbcSource.getColumnMetadata` for an example : https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/AbstractJdbcSource.java

Alternatively, you can try running all of the integration tests for the sources that inheirit from this class to see if it leads to an NPE since this codepath is covered in every integration test

Reply:
Got it. Good call. I'm still manually testing MySQL, but I'll run integration tests for less used connectors.